metals/src/main/scala/scala/meta/internal/metals/CodeLensProvider.scala

```diff
@@ -70,22 +103,60 @@ final class DebugCodeLensProvider(
           command <- commands
         } yield new l.CodeLens(range, command, null)
       case _ =>
-        Nil
+        Seq.empty
     }
   }
+
+  private def createSuperMethodCommand(
+      docWithPath: TextDocumentWithPath,
+      symbol: String,
+      role: SymbolOccurrence.Role,
+      cache: LensGoSuperCache,
+      findSymbol: String => Option[SymbolInformation]
+  ): Option[l.Command] = {
+    for {
+      symbolInformation <- findSymbol(symbol)
+      gotoLocation <- superMethodProvider.findSuperForMethodOrField(
+        symbolInformation,
+        docWithPath,
+        role,
+        findSymbol,
+        cache
+      )
+    } yield convertToSuperMethodCommand(
+      gotoLocation,
+      symbolInformation.displayName
+    )
+  }
+
+  private def convertToSuperMethodCommand(
+      gotoLocation: l.Location,
+      name: String
+  ): l.Command = {
+    new l.Command(
+      s"${config.icons.findsuper} ${name}",
+      ClientCommands.GotoLocation.id,
+      singletonList(gotoLocation)
+    )
+  }
 }
 
 object CodeLensProvider {
   import scala.meta.internal.metals.JsonParser._
 
+  type LensGoSuperCache =
```

Comment:
This is not a good place to keep the cache. The cache needs to be dropped on each semanticDB update, so that everything stays up to date and let's not cache entire TextDocuments :fearful: 

What we would really need is just an information if a symbol has a parent, GOTO Parent will calculate the rest.

Reply:
this is per request cache ...
If you have a class X with 20 methods you don't want for each method to calculate hierarchy of X.